### PR TITLE
AWS CI: Add Jenkinsfiles to ofiwg/libfabric

### PR DESCRIPTION
**Pull request by a-szegel**
The Jenkins multi-branch pipeline requires the Jenkinsfile to live in the upstream repo.

Testing evidence is here https://github.com/aws-ofiwg-bot/libfabric/pull/1, uncommented stuff because I was able to get the AWS accounts limits raised

### CODE REVIEW

File: .aws_ci/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            steps {
+                checkout scm
+            }
+        }
+        stage("Download and extract PortaFiducia") {
+            steps {
+                script {
+                    sh 'printenv'
+                    def common = load ".aws_ci/common.groovy"
+                    common.download_and_extract_portafiducia('PortaFiducia')
```
Comment (shijin-aws) [inline .aws_ci/Jenkinsfile:37]: is it something we want to disclose?

Comment (a-szegel) [reply .aws_ci/Jenkinsfile:37]: We have to in order to use the multibranch pipeline... I was able to hide the AWS account

**Review by shijin-aws — APPROVED** [review]

**Comment by a-szegel** [comment]
No reason to wait for AWS CI/Intel CI b/c we aren't modifying any files that matter yet.

**Comment by a-szegel** [comment]
@j-xiong Are you ok if I merge?

**Comment by shijin-aws** [comment]
Please add license/copyright headers for Amazon

**Comment by a-szegel** [comment]
done

**Comment by shijin-aws** [comment]
1 more comment: Should we store it in `contrib/aws` like cray and intel

**Comment by a-szegel** [comment]
done

**Review by shijin-aws — APPROVED** [review]

**Review by j-xiong — APPROVED** [review]

**Pull request merged**